trino/trino.go

```diff
@@ -1043,6 +1105,18 @@ func (qr *driverRows) initColumns(qresp *queryResponse) error {
 	return nil
 }
 
+func (qr *driverRows) updateCallback(c chan QueryProgressCallbackResponse) {
+	qrStats := <-c
+	currentTime := time.Now()
+	diff := currentTime.Sub(qr.stmt.conn.callbackPeriod.LastCallbackTime)
+	period := qr.stmt.conn.callbackPeriod.Period
+	// Check if period has passed OR if query state changes
+	if diff >= period || qrStats.LastQueryState != qrStats.QueryStats.State {
```

Comment:
Comparing if the state changed shouldn't be necessary. Stats include query progress, so you'd update the progress multiple times when the state is `RUNNING` and doesn't change.

Reply:
I included comparing state because if period hasn't passed, and state changes from RUNNING -> FINISHING/FINISHED and time take for finishing was less than period given,
last callback response would be running (does not return finish).

Reply:
I totally missed the comment, sorry!